### ISSUE TITLE
Vertical metrics for tall/deep glyphs in Windows environment / Microsoft products

### ISSUE THREAD

**Issue by chrissimpkins** [googlefonts/googlesans#435]
Builds @ https://github.com/googlefonts/googlesans/pull/416/commits/956c8b0836dd84b7416ee8b33e6c132df8bf1cb4

MS Word (Office 18.2110.13110.0) on Win 10 tested with Google Sans regular weight via the VF builds.

The clipping is influenced by the vertical line "Spacing" settings options under the Layout menu.

With 0pt before and 0pt after we see clipping:

![2021-12-22 12_09_15-Document docx - Word](https://user-images.githubusercontent.com/4249591/147130759-00d428a3-58c7-46db-a634-ced9ee7ebd18.png)

With 18pt before and 12pt after we see no clipping:

![2021-12-22 12_19_43-Document docx - Word](https://user-images.githubusercontent.com/4249591/147131025-c2d1b29b-c1c7-4755-9e4b-6450cf9b2d9b.png)

I thought that they would collide across overlapping text runs, but apparently Word handles this with clipping (or we aren't properly calculating the yMax / yMin metrics for the Win Asc / Win Desc settings).  Irrespective of the env specific approach to overlapping contours, I think that a *user* will want to make this line spacing adjustment to properly view multi-line text runs in Lao and other tall/deep char sets.  

I'm not sure that there is any other modification that we can make to address this.  Increasing global line spacing metrics across all scripts to accommodate large/deep glyphs in new merged scripts is not an option.  This may be a user detail for guidance documentation.  In contrast to many typefaces, this one is highly unlikely to find broad use in MS products.  If we do hear about this issue, I think that the next step is to consider Noto style script-specific typeface families with custom typo / hhea metrics definitions for the tall/deep areas of Google Sans.

cc @madig @belluzj @cadsondemakoffice 

cc @davelab6

_Originally posted by @chrissimpkins in https://github.com/googlefonts/googlesans/issues/416#issuecomment-999748412_

Moved this issue out of the Lao import PR thread so that we can track the response from Si Daniels and make any recommended adjustments to our metrics.

**Comment by chrissimpkins** [comment]
Waiting to hear back.  I followed up with Si today.

**Comment by chrissimpkins** [comment]
Recommendations from Si re: how to approach this issue in MS Word:

> 1) Require that users manually increase spacing whenever the font is used. Not great as it requires a lot of manual work to ensure text is rendering correctly and will lead to inconsistent line spacing (or overlooking clipping).
> 2) Split Lao and other ’tall’ scripts into a separate font with taller metrics that will allow typographically appropriate line heights for these scripts where clipping does not occur. This will result in more source management on Google’s side, but an overall better experience for users.

And a notable comment from Peter Constable:

> One thing important to note: Word is clipping in on-screen display, but not when printing. I believe they clip in on-screen display so that they don’t need to re-draw multiple lines when text is being edited to avoid artefacts from the edited line remaining in the adjacent lines. That was probably a perf issue in the 1990s, but I doubt it would be a perf issue today.
>
>I’m reminded of suggestions that have been made to be able to have per-script line metric data in a font. Possible; but we need engineers available that would implement support for such data in products.

**Comment by chrissimpkins** [comment]
@dberlow, John Hudson, and I just had a discussion about script-specific v metrics and the spec yesterday. The problem is going to be even more significant for tall/deep scripts in the GS Flex project, and may determine how far some scripts can go on the axes.

Thoughts about addressing this at the spec level with support for script-specific v metrics definitions @davelab6 @rsheeter?

**Comment by dberlow** [comment]
That was an interesting discussion.

My alignment instructions to John and all others, are to make your design for users of that script only, to compose by default without effort, while keeping all the major glyph groups within the em. This will force into view any required custom alignment axes for em-square sharing among scripts, and inter-script design space relocation, into programmable solutions. I.E. I think type designers need the Universal Em as a barrier and find the correct ways and places not to break out of it without leaving a programmable variable behind as evidence. 

John’s suggestion￼, Centers around the understandable opinion as something of a latinizer thinking in a world of instances. He stated that he cannot make the parts of Myanmar that align near Latin lowercase￼ do so, without extensive trespass beyond the em, and this is additionally justified, and here as well, because Indian scripts stack out of the em. 

My main reason for not wanting to mix like John, i.e. Latinizing at the expense of the em, is that the em is one thing to all scripts, and if you push the alignments around for one combo, you’re affecting size, and causing detrimental size effects to other scripts, and programs! In wght, we’ve shown that starting from an equal luminance among styles with common os/2 wght values, can simplify adjustments to luminance — And we have a grade axis to make corrections.

In alignments, my instinct says to make them all work within a single vertical space, and develop axes and composition from there. Again — we have a grade/weight axis to make stylistic corrections. That implies with a proper design space architecture, all scripts will work together in default mode, though no ideally, but from a single token input, engineering can make small programmable changes to size and grade as script-among-script harmonization requires, and the defaults work fine for any the script used alone with whatever traditional line spacing is required.

If you choose to say, "This Em confinement won't work with these scripts over here, (that are set in horizontal lines with vertically stacking words)", I'm pretty sure you consign yourself to a scramble, where it's no possible way to avoid nearly unimaginable complexity all over the place, instead of in the right place, which is a variable font of imaginable complexity.

That written, I have already stated that my job is to watch and guide and as far as I can tell, Either Way it's done, if the designer keeps the font in the Em, or Latinizes it to the lowercase and blows up the Em, FB is prepared to:

A. Make the Latin axes needed to work with all the size, weights and widths of all the scripts Google requires, 
B. Explore the parametric/programmatic requirements to go back and forth between fitting in the Em and working with Latin l.c.
C. Do what I described with whomever wants to cooperate.
D. Guide whomever doesn't want to do it that way.
